test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeTests.java

```diff
@@ -397,7 +397,116 @@ static public void testWithFieldOnNonExistentClass() throws Throwable {
 	 *
 	 *	Assert.assertFalse((refType != refType), "An identity (!=) comparison on the same refType should always return false");
 	 *	}
-	 */	
+	 */
+
+    /*    
+	 * Test monitorEnter on valueType
+	 * 
+	 * class TestMonitorEnterOnValueType {
+	 *  long longField
+	 * }
+	 */
+	@Test(priority=2)
+	static public void testMonitorEnterOnValueType() throws Throwable {
+		int x = 0;
+		int y = 0;
+		Object valueType = makePoint2D.invoke(x, y);
+
+		String fields[] = {"longField:J"};
+		Class<?> testMonitorEnterOnValueType = ValueTypeGenerator.generateRefClass("TestMonitorEnterOnValueType", fields);
+		MethodHandle monitorEnterOnValueType = lookup.findStatic(testMonitorEnterOnValueType, "testMonitorEnterOnObject", MethodType.methodType(void.class, Object.class));
+		try {
+			monitorEnterOnValueType.invoke(valueType);
+			Assert.fail("should throw exception. MonitorEnter cannot be used with ValueType");
+		} catch (IllegalMonitorStateException e) {}
+	}
+
+	/*
+	 * Test monitorExit on valueType
+	 * 
+	 * class TestMonitorExitOnValueType {
+	 *  long longField
+	 * }
+	 */
+	@Test(priority=2)
+	static public void testMonitorExitOnValueType() throws Throwable {
+		int x = 1;
+		int y = 1;
+		Object valueType = makePoint2D.invoke(x, y);
+
+		String fields[] = {"longField:J"};
+		Class<?> testMonitorExitOnValueType = ValueTypeGenerator.generateRefClass("TestMonitorExitOnValueType", fields);
+		MethodHandle monitorExitOnValueType = lookup.findStatic(testMonitorExitOnValueType, "testMonitorExitOnObject", MethodType.methodType(void.class, Object.class));
+		try {
+			monitorExitOnValueType.invoke(valueType);
```

Comment:
Were you able to confirm that the NLS message is printed properly with the className etc?

Reply:
yes, I checked it before. The NLS message is printed.